hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -479,6 +510,200 @@ public Hashtable<String, String> getPathStatus(final Path path,
+   * @throws AzureBlobFileSystemException exception thrown from
+   * {@link AbfsClient#getBlobProperty(Path, TracingContext)} call
+   */
+  private BlobProperty getBlobProperty(Path blobPath,
+      TracingContext tracingContext) throws AzureBlobFileSystemException {
+    AbfsRestOperation op = client.getBlobProperty(blobPath, tracingContext);
+    BlobProperty blobProperty = new BlobProperty();
+    final AbfsHttpOperation opResult = op.getResult();
+    blobProperty.setIsDirectory(opResult
+        .getResponseHeader(X_MS_META_HDI_ISFOLDER) != null);
+    blobProperty.setUrl(op.getUrl().toString());
+    blobProperty.setCopyId(opResult.getResponseHeader(X_MS_COPY_ID));
+    blobProperty.setPath(blobPath);
+    blobProperty.setCopySourceUrl(opResult.getResponseHeader(X_MS_COPY_SOURCE));
+    blobProperty.setStatusDescription(
+        opResult.getResponseHeader(X_MS_COPY_STATUS_DESCRIPTION));
+    blobProperty.setCopyStatus(getCopyStatus(opResult));
+    blobProperty.setContentLength(
+        Long.parseLong(opResult.getResponseHeader(CONTENT_LENGTH)));
+    return blobProperty;
+  }
+
+  @org.apache.hadoop.classification.VisibleForTesting
+  String getCopyStatus(final AbfsHttpOperation opResult) {
+    return opResult.getResponseHeader(X_MS_COPY_STATUS);
+  }
+
+  /**
+   * Call server API for ListBlob on the blob endpoint. This API returns a limited
+   * number of blobs and provide a field called as NextMarker which is reference to
+   * next list of blobs for the query. Server expects that the client calls this API
+   * in loop with the NextMarker received in previous iteration of backend call for
+   * the same request.
+   *
+   * @param sourceDirBlobPath path from where the list of blob is requried.
+   * @param tracingContext object of {@link TracingContext}
+   * @param maxPerServerCallResult define how many blobs can client handle in server response.
+   * In case maxResult <= 5000, server sends number of blobs equal to the value. In
+   * case maxResult > 5000, server sends maximum 5000 blobs.
+   * @param maxResult defines maximum blobs the method should process
+   * @param absoluteDirSearch defines if (true) it is blobList search on a
+   * definitive directory, if (false) it is blobList search on a prefix.
+   * @return List of blobProperties
+   *
+   * @throws AbfsRestOperationException exception from server-calls / xml-parsing
+   */
+  public List<BlobProperty> getListBlobs(Path sourceDirBlobPath,
+      TracingContext tracingContext, Integer maxPerServerCallResult,
+      final Integer maxResult, final Boolean absoluteDirSearch)
+      throws AzureBlobFileSystemException {
+    List<BlobProperty> blobProperties = new ArrayList<>();
+    String nextMarker = null;
+    do {
+      AbfsRestOperation op = client.getListBlobs(sourceDirBlobPath,
+          tracingContext, nextMarker, null, maxPerServerCallResult,
+          absoluteDirSearch);
+      BlobList blobList = op.getResult().getBlobList();
```

Comment:
hasResult is required only in the case when recovering from exception thrown from abfsHttpOperation, abfsRestOperation. Here we are not in the catch block. AbfsRestOperation will always have result, as other wise there will be an exception thrown wherein processResponse is not completed, for which code will never reach this line. it will return back from
`op = client.getListBlobs(sourceDirBlobPath,
          tracingContext, nextMarker, null, maxPerServerCallResult,
          absoluteDirSearch);`